File: TP1/PetShop.java
import java.util.ArrayList;
import java.util.Scanner;

class PetShop {
    // Atribut
    private ArrayList<Produk> daftarProduk;

    // Constructor
    public PetShop() {
        daftarProduk = new ArrayList<>();
        // Menambahkan 2 produk default saat objek PetShop dibuat
        daftarProduk.add(new Produk(1, "Makanan Kucing", "Makanan", 50000));
        daftarProduk.add(new Produk(2, "Kandang Anjing", "Aksesoris", 200000));
    }

    // Destructor (tidak diperlukan di Java, karena Java memiliki garbage
    // collection)

    // Method untuk menampilkan semua data
    public void tampilkanData() {
        if (daftarProduk.isEmpty()) {
            System.out.println("Tidak ada data produk.");
        } else {
            System.out.println("Daftar Produk di PetShop:");
            for (Produk produk : daftarProduk) {
                produk.tampilkanInfo(); // Menggunakan metode baru
            }
        }
    }

    // Method untuk menambahkan data baru
    public void tambahData(int id, String nama, String kategori, double harga) {
        Produk produkBaru = new Produk(id, nama, kategori, harga);
        daftarProduk.add(produkBaru);
        System.out.println("Data produk berhasil ditambahkan.");
    }

    // Method untuk mengubah data yang sudah ada
    public void ubahData(int id, String namaBaru, String kategoriBaru, double hargaBaru) {
        for (Produk produk : daftarProduk) {
            if (produk.getId() == id) {
                produk.setNama(namaBaru);
                produk.setKategori(kategoriBaru);
                produk.setHarga(hargaBaru);
                System.out.println("Data produk berhasil diubah.");
                return;
            }
        }
        System.out.println("Produk dengan ID " + id + " tidak ditemukan.");
    }

    // Method untuk menghapus data
    public void hapusData(int id) {
        for (Produk produk : daftarProduk) {
            if (produk.getId() == id) {
                daftarProduk.remove(produk);
                System.out.println("Data produk berhasil dihapus.");
                return;
            }
        }
        System.out.println("Produk dengan ID " + id + " tidak ditemukan.");
    }

    // Method untuk mencari data berdasarkan nama produk
    public void cariData(String nama) {
        boolean ditemukan = false;
        for (Produk produk : daftarProduk) {
            if (produk.getNama().equalsIgnoreCase(nama)) {
                System.out.println("Produk ditemukan:");
                produk.tampilkanInfo(); // Menampilkan dengan metode baru
                ditemukan = true;
            }
        }
        if (!ditemukan) {
            System.out.println("Produk dengan nama \"" + nama + "\" tidak ditemukan.");
        }
    }    

    public static class Produk {
        private int id;
        private String nama;
        private String kategori;
        private double harga;

        // Constructor
        public Produk(int id, String nama, String kategori, double harga) {
            this.id = id;
            this.nama = nama;
            this.kategori = kategori;
            this.harga = harga;
        }

        // Getter dan Setter
        public int getId() {
            return id;
        }

        public String getNama() {
            return nama;
        }

        public void setNama(String nama) {
            this.nama = nama;
        }

        public String getKategori() {
            return kategori;
        }

        public void setKategori(String kategori) {
            this.kategori = kategori;
        }

        public double getHarga() {
            return harga;
        }

        public void setHarga(double harga) {
            this.harga = harga;
        }

        public void tampilkanInfo() {
            System.out.println("ID: " + id);
            System.out.println("Nama: " + nama);
            System.out.println("Kategori: " + kategori);
            System.out.println("Harga: Rp" + harga);
            System.out.println("--------------------------");
        }
    }
}

File: python/PetShop.py
class PetShop:
    def __init__(self):
        self.daftar_produk = []
        # Menambahkan 2 produk default saat objek PetShop dibuat
        self.tambah_data(1, "Makanan Kucing", "Makanan", 50000)
        self.tambah_data(2, "Kandang Anjing", "Aksesoris", 200000)

    def tampilkan_data(self):
        if not self.daftar_produk:
            print("Tidak ada data produk.")
        else:
            print("Daftar Produk di PetShop:")
            for produk in self.daftar_produk:
                self.tampilkan_info(produk)

    def tambah_data(self, id, nama, kategori, harga):
        produk_baru = {"id": id, "nama": nama, "kategori": kategori, "harga": harga}
        self.daftar_produk.append(produk_baru)
        print("Data produk berhasil ditambahkan.")

    def ubah_data(self, id, nama_baru, kategori_baru, harga_baru):
        for produk in self.daftar_produk:
            if produk["id"] == id:
                self.set_nama(produk, nama_baru)
                self.set_kategori(produk, kategori_baru)
                self.set_harga(produk, harga_baru)
                print("Data produk berhasil diubah.")
                return
        print(f"Produk dengan ID {id} tidak ditemukan.")

    def hapus_data(self, id):
        for produk in self.daftar_produk:
            if produk["id"] == id:
                self.daftar_produk.remove(produk)
                print("Data produk berhasil dihapus.")
                return
        print(f"Produk dengan ID {id} tidak ditemukan.")

    def cari_data(self, nama):
        ditemukan = False
        for produk in self.daftar_produk:
            if self.get_nama(produk).lower() == nama.lower():
                print("Produk ditemukan:")
                self.tampilkan_info(produk)
                ditemukan = True
        if not ditemukan:
            print(f'Produk dengan nama "{nama}" tidak ditemukan.')

    def tampilkan_info(self, produk):
        print(f"ID: {produk['id']}")
        print(f"Nama: {produk['nama']}")
        print(f"Kategori: {produk['kategori']}")
        print(f"Harga: Rp{produk['harga']}")
        print("--------------------------")

    # Getter
    def get_id(self, produk):
        return produk["id"]

    def get_nama(self, produk):
        return produk["nama"]

    def get_kategori(self, produk):
        return produk["kategori"]

    def get_harga(self, produk):
        return produk["harga"]

    # Setter
    def set_nama(self, produk, nama):
        produk["nama"] = nama

    def set_kategori(self, produk, kategori):
        produk["kategori"] = kategori

    def set_harga(self, produk, harga):
        produk["harga"] = harga
